controllers/storagecluster/storageclient.go

```diff
@@ -145,16 +143,27 @@ func (s *storageClient) updateClientConfigMap(r *StorageClusterReconciler, names
 
 // getCephNetworkAnnotationValue returns the network annotation value for the given StorageCluster NetworkSpec.
 func getCephNetworkAnnotationValue(cephNetworkSpec *rookCephv1.NetworkSpec, scNamespace string) (string, error) {
+	if cephNetworkSpec == nil {
+		return "", nil // cannot be multus if no network spec
+	}
+	if !cephNetworkSpec.IsMultus() {
+		return "", nil // if not multus, no annotation to add
+	}
 	if len(cephNetworkSpec.Selectors) == 0 {
 		return "", fmt.Errorf("invalid ceph network spec")
 	}
+
 	networkSelectionElement, err := cephNetworkSpec.GetNetworkSelection(scNamespace, rookCephv1.CephNetworkType("public"))
-	if err != nil || networkSelectionElement == nil {
+	if err != nil {
 		return "", fmt.Errorf("failed to get network selection element: %v", err)
 	}
-	nwAnnotation, err := json.Marshal([]*nadv1.NetworkSelectionElement{networkSelectionElement})
+	if networkSelectionElement == nil {
+		return "", nil // annotation only needed for clusters w/ multus public net selected
+	}
+
+	nwAnnotation, err := rookCephv1.NetworkSelectionsToAnnotationValue(networkSelectionElement)
```

Comment:
One of the other changes I made was to begin using Rook's helper function designed to convert network selections to annotation values. While marshalling to the struct is essentially all that is needed, usage of Rook's helper function allows ocs-operator to benefit from Rook's unit and CI testing of the functionality.

QE may need to note that the format is expected to be in this format: `[{"name":"odf-public","namespace":"ns"}]` including the surrounding square brackets `[]`.

If anyone disagrees with the benefit, feel free to revert this to JSON marshalling as desired.